Sources/PenguinParallel/NonblockingThreadPool/NumberOperations.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Returns an array of all positive integers that are co-prime with `n`.
+///
+/// Two numbers are co-prime if their GCD is 1.
+internal func makeCoprimes(upTo n: Int) -> [Int] {
+  var coprimes = [Int]()
+  for i in 1...n {
+    var a = i
+    var b = n
+    // If GCD(a, b) == 1, then a and b are coprimes.
+    while b != 0 {
+      let tmp = a
+      a = b
+      b = tmp % b
+    }
+    if a == 1 { coprimes.append(i) }
+  }
+  return coprimes
+}
+
+/// Reduce `lhs` into `[0, size)`.
```

Comment:
Form problems: 

1. this reads roughly like the summary for a thing with side-effects.

2. If you're going to use the backticks around the whole thing, you should use Swift range notation, so
    ```suggestion
    /// Reduce `lhs` into [0, `size`).
    ```
    or 

    ```suggestion
    /// Reduce `lhs` into `0..<size`.
    ```

More importantly, the summary is meaningless to me.  I had to read the paragraph below to even get a hint as to what it was doing, and follow the link to actually learn.  

```suggestion
/// Returns a value deterministically selected from `0..<size` 
/// with uniform probability over possible values of `lhs`. 
```

Reply:
Yeah, that's definitely much nicer.

Reply:
Still disappointed you didn't take my whole summary.  In particular not to see `lhs` mentioned at all makes it hard to see what relationship it might have to the result.  Also can we name that thing something representative of the role, like “selector” or something?

Another question that comes to my mind: do I still get uniform distribution if, say, the high N bits of LHS are always zero?  In other words, is the result for lhs=k near or distant from the result for lhs=k+1?

Reply:
Oh, apologies for missing that, and for only seeing your comment now; https://github.com/saeta/penguin/pull/46 for the fix (LMK if I missed something else)!

For context, I often choose to re-type the suggestions out. I find it helps me internalize them and think about them more deeply. I will try and be more careful about not accidentally truncating them in the future!